dtests/src/test/java/io/snappydata/hydra/cluster/SnappyPrms.java

```diff
@@ -117,6 +122,11 @@
     public static Long logLevel;
 
     /**
+     * (String) String to be serached in the spark job output file in order to know the job completion status.
+     */
+    public static Long searchString;
```

Comment:
when the spark job is completed   it displays following string
Job $jobId finished: $callsite, took $time in seconds

when the spark job is failed  it displays following string
Job $jobId failed: $callsites,  took $time in seconds

Instead of asking user to provide a search String, you can look for the above strings. 


The other preferable way  could be to register for the  following listener (SparkListener) with  the SparkContext and write the code there for job completion handling.

@DeveloperApi
case class SparkListenerJobEnd(
    jobId: Int,
    time: Long,
    jobResult: JobResult)
  extends SparkListenerEvent